internal/test/server.go

```diff
@@ -124,9 +125,9 @@ func RequireFakeEnvoyTarGz(t *testing.T, v version.PatchVersion) ([]byte, versio
 	require.NoError(t, err)
 
 	// Read the tar.gz into a byte array. This allows the mock server to set content length correctly
-	f, err := os.Open(tempGz) //nolint:gosec
+	f, err := os.Open(tempGz)
 	require.NoError(t, err)
-	defer f.Close() // nolint
+	t.Cleanup(func() { require.NoError(t, f.Close()) })
```

Comment:
this one is worse than before. `RequireFakeEnvoyTarGz` isn't a test method, it is a utility, so cleanup on exit is valid and also less crufty.